config/config_mgmt.py

```diff
@@ -213,6 +220,70 @@ def writeConfigDB(self, jDiff):
 
         return
 
+    def add_module(self, yang_module_str, replace_if_exists=False):
+        """
+        Validate and add new YANG module to the system.
+
+        Parameters:
+            yang_module_str (str): YANG module in string representation.
+
+        Returns:
+            None
+        """
+
+        module_name = self.get_module_name(yang_module_str)
+        module_path = os.path.join(YANG_DIR, '{}.yang'.format(module_name))
+        if os.path.exists(module_path) and not replace_if_exists:
+            raise Exception('{} already exists'.format(module_name))
+        with open(module_path, 'w') as module_file:
+            module_file.write(yang_module_str)
+        try:
+            self.__init_sonic_yang()
+        except Exception:
+            os.remove(module_path)
+            raise
+
+    def remove_module(self, module_name):
+        """
+        Remove YANG module on the system and validate.
+
+        Parameters:
+            module_name (str): YANG module name.
+
+        Returns:
+            None
+        """
+
+        module_path = os.path.join(YANG_DIR, '{}.yang'.format(module_name))
+        if not os.path.exists(module_path):
+            return
+        with open(module_path, 'r') as module_file:
+            yang_module_str = module_file.read()
```

Comment:
Instead can we rename the file as {module_path}.xyz; call _init_sonic_yang and upon success remove the renamed file and upon failure rename back to module_path ?

Reply:
I changed it the way you advised. Although, instead of rename() I used shutil.move() which will rename() in case /tmp/ is on the same filesystem, otherwise use delete and copy.